samples/testbot/test-bot/BotOrchestration.cs

```diff
@@ -0,0 +1,379 @@
+                }
+            }
+            else
+            {
+                // Create a new function app
+                if (!await TryCallDeploymentServiceHttpApiAsync("api/CreateNewFunctionApp", context, log, testParameters))
+                {
+                    string message = $"Failed to create a new function app! 💣 Check the internal deployment service logs for more details.";
+                    await context.CallActivityAsync(nameof(PatchGitHubComment), (testParameters.GitHubCommentIdApiUrl, message));
+                    throw new Exception(message);
+                }
+                else
+                {
+                    await context.CallActivityAsync(nameof(PatchGitHubComment), (testParameters.GitHubCommentIdApiUrl, "Successfully created a new function app."));
+                }
+            }
+
+            try
+            {
+                // Deploy and start the test app
+                HttpManagementPayload managementUrls = null;
+                if (!await TryCallDeploymentServiceHttpApiAsync(
+                    "api/DeployToFunctionApp",
+                    context,
+                    log,
+                    testParameters,
+                    (responseJson) => managementUrls = JsonConvert.DeserializeObject<HttpManagementPayload>(responseJson)))
+                {
+                    string message = $"Failed to deploy the test app! 💣 Check the internal deployment service logs for more details.";
+                    await context.CallActivityAsync(nameof(PatchGitHubComment), (testParameters.GitHubCommentIdApiUrl, message));
+                    throw new Exception(message);
+                }
+
+                // Get the URL for doing status queries
+                if (managementUrls == null || string.IsNullOrEmpty(managementUrls.StatusQueryGetUri))
+                {
+                    string message = $"The deployment service API call succeeded but returned an unexpected response. Check the logs for details.";
+                    await context.CallActivityAsync(nameof(PatchGitHubComment), (testParameters.GitHubCommentIdApiUrl, message));
+                    throw new Exception(message);
+                }
+
+                DurableOrchestrationStatus status;
+                try
+                {
+                    status = await WaitForStartAsync(context, log, managementUrls);
+                }
+                catch(Exception)
+                {
+                    throw;
+                }
+                
+                if (status == null)
+                {
+                    string message = $"The test was scheduled but still hasn't started! Giving up. 😞";
+                    await context.CallActivityAsync(nameof(PatchGitHubComment), (testParameters.GitHubCommentIdApiUrl, message));
+                    throw new Exception(message);
+                }
+
+                string previousCustomStatus = string.Empty;
+                while (true)
```

Comment:
we may want to have a cutoff time so we don't end up infinitely polling if something goes horribly wrong.